### PR TITLE
Add autoapproval for some types of name deletion requests

### PR DESCRIPTION
**Pull request by emily-grabowski**

### CODE REVIEW

File: openreview/profile/process/request_remove_name_process.py
Comment (carlosmondra) [inline openreview/profile/process/request_remove_name_process.py:37]: Do we need to check publications after it has been approved? Shouldn't we just return in this case?

File: openreview/profile/process/request_remove_name_process.py
Comment (carlosmondra) [inline openreview/profile/process/request_remove_name_process.py:35]: Maybe we should return after the name has been approved. I don't think we should check for publications.

Comment (emily-grabowski) [reply openreview/profile/process/request_remove_name_process.py:35]: Checking for publications was something that was already in the process function before my changes -  the name would be autoapproved if it did not appear in any publications, so I left the code in for that. I'm not sure whether we want to keep that system, since it means people can make pretty drastic changes to their name as long as its not associated with a submission.

Comment (carlosmondra) [reply openreview/profile/process/request_remove_name_process.py:35]: I'm not sure if I didn't understand or if I phrased my comment incorrectly.

So suppose that you auto approve the name change because it passes your validation. The name will be changed, but it will still send an email to the user if it has publications telling him that we are reviewing his request. Is this what we want?

Comment (celestemartinez) [reply openreview/profile/process/request_remove_name_process.py:35]: Should we do this validation then only if no publications are found that use this username?

Comment (emily-grabowski) [reply openreview/profile/process/request_remove_name_process.py:35]: @carlosmondra Got it that makes sense- with the way it is currently written it will accept the name deletion request, but still send an email saying that we are reviewing the changes if there are publications - I'll add a return to prevent that

Comment (emily-grabowski) [reply openreview/profile/process/request_remove_name_process.py:35]: Celeste, could you clarify? I'm not sure I follow

Comment (celestemartinez) [reply openreview/profile/process/request_remove_name_process.py:35]: I guess I am just wondering if we should add this check to automatically approve after we have checked if there are publications. It looks like if there are publications, we always send the email about evaluating the request. 
I am not sure if it's correct to automatically approve some cases before we check if the user has publications?

Comment (emily-grabowski) [reply openreview/profile/process/request_remove_name_process.py:35]: The original code would autoapprove all requests that had no publications and moderate all requests associated with publications. 

The change I proposed was to reduce moderation by also auto-approving certain simple changes (capitalization, spacing, name reversal) that are associated with publications. Does this seem like a reasonable change to you? I'm happy to discuss further if you think it'll cause any issues.

Comment (celestemartinez) [reply openreview/profile/process/request_remove_name_process.py:35]: ohh I understand now, thanks for clarifying! Your change looks good.

**Review by carlosmondra — APPROVED** [review]
Awesome!

**Pull request merged**
